diff --git a/code_quality.sh b/code_quality.sh
index 3083171..237054c 100755
--- a/code_quality.sh
+++ b/code_quality.sh
@@ -2,21 +2,36 @@
 
 set -e
 
-# Run flake8, pylint, mypy and capture their exit statuses
+overall_status=0
+
 flake8 ./
 flake8_status=$?
+if [[ $flake8_status -ne 0 ]]; then
+  overall_status=1
+fi
 
 pylint ./
 pylint_status=$?
+if [[ $pylint_status -ne 0 ]]; then
+  overall_status=1
+fi
 
 mypy app/
 mypy_status=$?
+if [[ $mypy_status -ne 0 ]]; then
+  overall_status=1
+fi
 
 bandit -r app/
 bandit_status=$?
+if [[ $bandit_status -ne 0 ]]; then
+  overall_status=1
+fi
 
-# Fail the job if any tool failed
-if [[ $flake8_status -ne 0 || $pylint_status -ne 0 || $mypy_status -ne 0 || $bandit_status -ne 0 ]]; then
+if [[ $overall_status -ne 0 ]]; then
   echo "Code quality checks failed."
   exit 1
-fi
+else
+  echo "All code quality checks passed!"
+  exit 0
+fi
\ No newline at end of file
